File: Longest_Subsequence/longest.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @returns {number[]}
 */
var answerQueries = function (nums, queries) {
	const result = new Array(queries.length);

	nums.sort((a, b) => a - b);
	for (let j = 0; j < queries.length; j++) {
		let [sum, subsequenceCount] = [0, 0];

		for (let i = 0; i < nums.length; i++) {
			if (sum + nums[i] <= queries[j]) {
				sum += nums[i];
				subsequenceCount++;
			} else {
				break;
			}
		}
		result[j] = subsequenceCount;
	}

	return result;
};

File: Longest_Subsequence/longest.cs
using System;

public class Solution
{
    public int[] AnswerQueries(int[] nums, int[] queries)
    {
        Array.Sort(nums);
        int[] result = new int[queries.Length];

        for (int i = 0; i < queries.Length; i++)
        {
            int sum = 0,
                subsequenceCount = 0;

            for (int j = 0; j < nums.Length; j++)
            {
                if (sum + nums[j] <= queries[i])
                {
                    sum += nums[j];
                    subsequenceCount++;
                }
                else
                {
                    break;
                }
            }
            result[i] = subsequenceCount;
        }

        return result;
    }
}
